{{cookiecutter.project_slug}}/.gitlab-ci.yml

```diff
@@ -0,0 +1,131 @@
+# templates
+# ------------
+
+.deploy-template:
+  image: alpine:latest
+  script:
+    - echo "Deploying to ${CI_ENVIRONMENT_NAME}"
+  variables:
+    GIT_STRATEGY: none  # prevents fetching repo (we only need the artifact here)
+
+# -------------
+# Pipeline Definition
+
+variables:
+  PIP_CACHE_DIR: $CI_PROJECT_DIR/.cache/pip
+  {%- if cookiecutter.package_manager == 'conda' %}
+  CONDA_PKGS_DIRS: $CI_PROJECT_DIR/.cache/conda
+  {%- elif cookiecutter.package_manager == 'poetry' %}
+  POETRY_VERSION: 1.0.5
+  POETRY_CACHE_DIR: $CI_PROJECT_DIR/.cache/poetry
+  {%- endif %}
+
+cache: &python_cache
+  paths:
+    - $PIP_CACHE_DIR
+    {%- if cookiecutter.package_manager == 'conda' %}
+    - $CONDA_PKGS_DIRS/*.tar.bz2
+    - $CONDA_PKGS_DIRS/urls.txt
+    {%- elif cookiecutter.package_manager == 'poetry' %}
+    - $POETRY_CACHE_DIR
+    {%- endif %}
+
+stages:
+  - build-wheel
+  - test-unit
+  - deploy-nonprod
+  - deploy-prod
+
+# Build and Test
+{% if cookiecutter.package_manager == 'poetry' -%}
+build-wheel:
+  image: python:3.7-stretch
```

Comment:
let's use `python:3-stretch` instead, so we always get the latest version of Python 3. No need to limit ourselves to 3.7, the current stable version is 3.9 and this will change frequently

Reply:
I agree. We either way install our python version from the dependencies. Will change that.